tests/README.md

```diff
@@ -59,62 +59,50 @@ expect(onyxData).toBe(expectedOnyxData);
+  - "When" - Describes what action is being done and the thing that is being tested
+  - "Then" - Describes what is expected to happen
 
-// When <something_happens>
-... code that does something
+2. DO explain **WHY** the test is doing what it is doing in every comment.
+3. DO NOT explain **WHAT** the code is doing in comments. This information should be self-evident.
+
+The format looks like this:
 
-// Then <expectation>
-... code that performs the assertion
 ```
+// BAD
+// Given an account
+{* code that sets initial condition *}
 
-## Example Test
+// When it is closed
+{* code that does something *}
 
-```javascript
-HttpUtils.xhr = jest.fn();
-
-describe('actions/Report', () => {
-    it('adds an optimistic comment', () => {
-        // Given an Onyx subscription to a report's `reportActions`
-        const ACTION_ID = 1;
-        const REPORT_ID = 1;
-        let reportActions;
-        Onyx.connect({
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${REPORT_ID}`,
-            callback: val => reportActions = val,
-        });
-
-        // Mock Report_AddComment command so it can succeed
-        HttpUtils.xhr.mockImplementation(() => Promise.resolve({
-            jsonCode: 200,
-        }));
-
-        // When we add a new action to that report
-        Report.addComment(REPORT_ID, 'Hello!');
-        return waitForBatchedUpdates()
-            .then(() => {
-                const action = reportActions[ACTION_ID];
-
-                // Then the action set in the Onyx callback should match
-                // the comment we left and it will be in a loading state because
-                // it's an "optimistic comment"
-                expect(action.message[0].text).toBe('Hello!');
-                expect(action.isPending).toBe(true);
-            });
-    });
-});
+// Then the user is logged out
+{* code that performs the assertion *}
+
+// GOOD
+// Given an account of a brand new user
+{* code that sets initial condition *}
+
+// When the account is closed by clicking on the close account button
+{* code that does something *}
+
+// Then the user should be logged out because their account is no longer active
+{* code that performs the assertion *}
 ```
 
 ## When to Write a Test
 
-Many of the UI features of our application should go through rigorous testing by you, your PR reviewer, and finally QA before deployment. It's also difficult to maintain UI tests when the UI changes often. Therefore, it's not valuable for us to place every single part of the application UI under test at this time. The manual testing steps should catch most major UI bugs. Therefore, if we are writing any test there should be a **good reason**.
+Many of the UI features of our application should go through rigorous testing by you, your PR reviewer, and finally QA before deployment. However, the code is mature enough now that protecting code against regressions is the top priority.
 
 **What's a "good reason" to write a test?**
```

Comment:
Lets remove this line so that you must have a good reason to NOT write a test

Reply:
And lets include these as a starting point: https://expensify.slack.com/archives/CC7NECV4L/p1731686459863719?thread_ts=1731609633.072659&cid=CC7NECV4L

Reply:
I kept both, but put the "skip reasons" first as I think its still useful info there